include/dlaf/common/callable_object.h

```diff
@@ -17,12 +17,14 @@
 /// require explicit selection of the overload when passing the function to
 /// higher-order functions. With this wrapper the overload is selected inside
 /// the wrapper's operator().
-#define DLAF_MAKE_CALLABLE_OBJECT(fname)                                                 \
-  constexpr struct fname##_t {                                                           \
-    template <typename... Ts>                                                            \
-    auto operator()(Ts&&... ts) const noexcept(noexcept(fname(std::forward<Ts>(ts)...))) \
-        -> decltype(fname(std::forward<Ts>(ts)...)) {                                    \
-      return fname(std::forward<Ts>(ts)...);                                             \
-    }                                                                                    \
-  } fname##_o {                                                                          \
+///
+/// TODO: Aaargh, (f)(...) turns off ADL!? That's great, but too obscure?
```

Comment:
This was an interesting one... The wrapper for `copy` would in the changes that I've made attempt to call `std::copy` because of ADL. Wrapping a function name in parentheses seems to be a valid way to disable ADL and that is exactly what we want here. However, is it too obscure? It definitely needs a comment if this will be the way we disable ADL. Other wrappers likely also need to disable ADL just to be on the safe side. The other option is qualifying the function name which is simple, but it may not be as obvious at the callsite (of the macro) why the qualified name is required.

Reply:
I moved this to #580 since it no longer seems to be required specifically for this PR, but I still find it a useful change.